insights/core/plugins.py

```diff
@@ -190,6 +191,29 @@ def register_cluster_reducer(func):
     p["module"] = func.__module__
 
 
+def deprecated(solution):
```

Comment:
@PaulWay how have you tested this?  I tried it with a class method and it appears to create an error for the entire class regardless of whether the method is used.

Reply:
@bfahr I tested this initially using the deprecated oracle parser functions from #356.  Do you think I need to put in tests for this?  That'd be a good idea, I suppose...

If we deprecated a class, then I think using the entire class should generate a deprecation error.  What do you think should happen?

Reply:
@PaulWay sorry I wasn't clear, I wasn't necessarily advocating adding tests, I was just wondering how you tested.  If a method in a class is deprecated then it seems that whether or not the method is used, if the class is used then it generates the error.  Is there a way to deprecate a class method and not have an error raised unless the class method is used?

Reply:
Hi @bfahr - ah, so you did something like this:

```python
class FlangParser(Parser):
    @deprecated("use new `search` method with key=value parameters")
    def old_search(self, name, mode, cpus, sequence, escalation, paradigm):
        ...
```
And the whole class got flagged?  I didn't test that.  Let me see what I can do.

Reply:
OK, one problem here is that the method I used above causes the mere existence of a deprecated function to cause an error.  Even if that function doesn't get used, the function's existence in the code causes the error.  What I want is for a warning to be generated only when the function gets used.  So I've rewritten the decorator as:

```python
    def _log_deprecated(func):
        message = "Warning: {mod}.{fn} has been deprecated: {sol}".format(
            mod=func.__module__, fn=func.__name__, sol=solution
        )

        def function_with_deprecation_warning(*args, **kwargs):
            warnings.warn(message, DeprecationWarning)
            return func(*args, **kwargs)

        return function_with_deprecation_warning

    return _log_deprecated
```

I'm fine-tuning its output for functions, classes and methods, but this AFAICS works better than deprecating the existence of the function.  What do you think, @bfahr and @jhjaggars ?

Reply:
@PaulWay this now seems to trigger on the method, but it still triggers when the method is not being used.  I just added a dummy method to test that.  Did I miss something?